Makefile

```diff
@@ -508,6 +508,11 @@ website-build:
 deploy-testremote-teardown:
 	bash "ci/deploy-testremote-teardown.sh"
 
+.PHONY: unit-tests
+unit-tests:
+	yarn --frozen-lockfile
```

Comment:
Okay... :) in the current setup _within_ the preamble we have a containerized environment where `node_modules` is set up through a big `yarn install` (_in_ the container file system) and practically the only build artifact of relevance that is carried over to other steps is the single-binary CLI build (through the "prebuild" directory).

Here, we're doing another biiig `yarn install` (_in_ the container file system) "just" for being able to run these unit tests. 

Certainly something to consolidate in the future.

What are your thoughts about running these unit tests as part of the preamble, like we currently do for the go unit tests?

(note1: not suggesting to change anything _now_ :)).

(note2: sharing a cache between these `yarn install`s failed rather miserably.)

Reply:
> Certainly something to consolidate in the future.

Well #119 helps with that already. The [build dir](https://github.com/opstrace/opstrace/blob/main/Makefile#L406) is available in the container so the `node_modules` are only installed once.

I'm waiting for the currents tests to finish since yesterday they triggered some flakes. I'll update this PR with the merge conflicts after they finish.

> What are your thoughts about running these unit tests as part of the preamble, like we currently do for the go unit tests?

I was thinking of following up with a PR to move the tests out of the preamble. The preamble step would check everything builds. The unit-tests step would run the (well) the unit tests. And then the next steps would continue as today. 

The point is to make it obvious to know know what's broken. It's the build. It's the unit tests. It's the integration/e2e tests. Etc.

Reply:
> The build dir is available in the container

Huh. I created this system and keep having wrong intuitions about what's happening :). Too complex.

> I was thinking of following up with a PR to move the tests out of the preamble. The preamble step would check everything builds. The unit-tests step would run the (well) the unit tests. And then the next steps would continue as today.

certainly a +1!

I forgot to say that before; a build job that is meant to run unit tests is certainly great; much better semantics than piggy-backing this into the preamble stage.

Big thanks for taking this on.